functions/pm_functions.scm

```diff
@@ -435,13 +457,111 @@
 		 ))
 	 )
 
-		(EvaluationLink
-		  (PredicateNode "interacts_with")
-		   (ListLink
-			   (VariableNode "$c")
-			   (VariableNode "$b")
-		))
+  ;; This will be executed if the above pattern is found.
+  (ExecutionOutputLink
+    (GroundedSchemaNode "scm: generate-result")
+		  (ListLink
+		    (VariableNode "$c")
+		    (VariableNode "$b")
+		  ))
 	
-	))
+))
+))
+
+;; Grounded schema node to add info about matched variable nodes
+
+(define (generate-result var1 var2)
```

Comment:
Should this function be in this file as it isn't doing any pattern matching? Better if it is moved to `utils.scm`